flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt

```diff
@@ -6,94 +6,163 @@
+          string.startsWith("[MyHomePage,") && string.endsWith("FloatingActionButton]")
+        }
+      }
+
+      // This is disabled due to an issue in the test framework. The #selectRow call causes
+      // the widget tree to change its selection, which is absolutely not what we want.
+      //      step("Details selection leaves main tree unchanged") {
+      //        val string = detailsTree.selectionSync().toString()
+      //        detailsTree.selectRow(1, expand = false)
+      //        pause(object : Condition("Details tree changes") {
+      //          override fun test(): Boolean {
+      //            return string != detailsTree.selectionSync().toString()
+      //          }
+      //        }, Timeouts.seconds05)
+      //        expect("[[root], MyApp, MaterialApp, MyHomePage, Scaffold, FloatingActionButton]") {
+      //          inspectorTree.selectionSync().toString()
+      //        }
+      //      }
+
       runner().stop()
     }
   }
 
-  private fun IdeFrameFixture.runner(): ExecutionToolWindowFixture.ContentFixture {
-    return runToolWindow.findContent("main.dart")
+  @Test
+  fun hotReload() {
+    ProjectCreator.importProject()
+    ideFrame {
+      launchFlutterApp()
+      val inspector = flutterInspectorFixture(this)
+      inspector.populate()
+      val widgets = inspector.widgetsFixture()
+      val widgetsTree = widgets.inspectorTreeFixture(isDetails = false)
+      widgetsTree.selectRow(0)
+      val detailsTree = widgets.inspectorTreeFixture(isDetails = true)
+      val initialDetails = detailsTree.selectionSync().toString()
+
+      editor {
+        // Wait until current file has appeared in current editor and set focus to editor.
+        moveTo(0)
+        val editorCode = getCurrentFileContents(false)!!
+        val original = "pushed the button this"
+        val index = editorCode.indexOf(original) + original.length
+        moveTo(index)
+        val key = if (isMac) KeyEvent.VK_BACK_SPACE else KeyEvent.VK_DELETE
+        for (n in 1..4) typeKey(key)
+        typeText("that")
+        //typeKey(KeyEvent.VK_ESCAPE) // Dismiss completion popup -- not needed with "that" but is needed with "so"
+      }
+
+      step("Trigger Hot Reload and wait for it to finish") {
+        val reload = findHotReloadButton()
+        reload.click()
+        editor.clickCenter() // Need to cycle the event loop to get button enabled on Mac.
+        pause(object : Condition("Hot Reload finishes") {
+          override fun test(): Boolean {
+            return reload.isEnabled
+          }
+        }, Timeouts.seconds05)
+        step("Work around #3370") {
+          // https://github.com/flutter/flutter-intellij/issues/3370
+          inspector.renderTreeFixture().show() // The refresh button is broken so force tree update by switching views.
+          inspector.populate()
+          widgets.show() // And back to the one we want
+        }
+        widgetsTree.selectRow(6) // Text widget
```

Comment:
Out of curiosity, is there a more semantic way of identifying the row to select?

Reply:
I considered a named constant here, but didn't bother, mostly because all the other embedded constants don't really need names. The sample code isn't generated by Flutter, so it won't change unless we want it to. If we change it, this test will break, and the comment should at least point to a solution.

Reply:
Sounds good.